api/preprints/serializers.py

```diff
@@ -250,7 +255,6 @@ def create(self, validated_data):
             raise Conflict('Only one preprint per provider can be submitted for a node. Check `meta[existing_resource_id]`.', meta={'existing_resource_id': node_preprints.first()._id})
 
         preprint = PreprintService(node=node, provider=provider)
-        self.set_field(preprint.set_primary_file, primary_file, auth)
```

Comment:
Doesn't this removal break backwards-compatibility the front-end flow, which currently passes a file with the preprint `POST`?

Reply:
Nope! The primary preprint file is instead set in the `update()` call -- which happens at the end of the create method. Tested locally on current frontend and works ok. The trade off is that the check for preprint file has to happen on publish instead of `create()`